sos/collector/__init__.py

```diff
@@ -1089,7 +1090,8 @@ def get_nodes(self):
                     self.node_list.remove(node)
             if not self.cluster.strict_node_list:
                 self.node_list.append(self.hostname)
-        self.reduce_node_list()
+        if not self.cluster.strict_node_list:
+            self.reduce_node_list()
```

Comment:
I think we need to keep this. It would be an edge case, but if nodes are somehow duplicated, this prevents us from trying to connect and collect from the node twice.

Reply:
Assume `pacemaker` cluster with nodes declared as IP addresses (for a reproducer, just have `/etc/corosync/corosync.conf` with one `ring0_addr:` pointing to your IP address and another `ring0_addr:` pointing to a different IP address - until you have `crm_mon` installed, then `sos collect --cluster-type pacemaker --batch -o qpid` will attempt to collect sosreport only from the other IP address but not from own).

Then, https://github.com/sosreport/sos/blob/main/sos/collector/__init__.py#L1029-L1031 will remove the IP address of the local host.

When having `ring0_addr: $(hostname -f)`, then https://github.com/sosreport/sos/blob/main/sos/collector/__init__.py#L1034-L1037 removes it from the node list, and the change from point 2. (`if self.primary.connected and not self.cluster.strict_node_list:`) will cause skipping sosreport from the local host.

Also not sure if similar scenario with hostname can't apply to https://github.com/sosreport/sos/blob/main/sos/collector/__init__.py#L1027-L1028 ? (I dont think so but unsure now)

While the rest of the `reduce_node_list` makes sense.

So I will update the PR to:
- call `reduce_node_list` unconditionally again
- within the method:
  -  call `for i in self.ip_addrs:` only when `not self.cluster.strict_node_list:`
  - enhance `if self.primary is not None:` by the same `and not self.cluster.strict_node_list:`

But still I feel there can be a use case where we keep/remove local host wrongly, as the conditions at https://github.com/sosreport/sos/blob/main/sos/collector/__init__.py#L1034 and https://github.com/sosreport/sos/blob/main/sos/collector/__init__.py#L1180 will differ?